app/assets/javascripts/discourse/widgets/user-menu.js.es6

```diff
@@ -81,21 +85,149 @@ createWidget('user-menu-links', {
   }
 });
 
+export default createWidget('user-bookmarks', {
+  buildKey: () => 'user-bookmarks',
+
+  defaultState() {
+    return { loading: false, loaded: false, content: [] };
+  },
+
+  refreshBookmarks(state) {
+    if (state.loading) { return; }
+
+    const { currentUser } = this;
+    let findUrl = Discourse.getURL('/user_actions.json?username=' + currentUser.username + "&filter=" + Discourse.UserAction.TYPES.bookmarks);
+
+    state.loading = true;
+    return ajax(findUrl, {cache: 'false'}).then(result => {
+      if (result && result.user_actions) {
+        const copy = Em.A();
+        result.user_actions.forEach(function(action) {
+          action.title = emojiUnescape(Handlebars.Utils.escapeExpression(action.title));
+          copy.pushObject(UserAction.create(action));
+        });
+        state.content.pushObjects(UserAction.collapseStream(copy));
+      }
+    }).finally(() => {
+      state.loading = false;
+      state.loaded = true;
+      this.scheduleRerender();
+    });
+  },
+
+  html(attrs, state) {
+    if (!state.loaded) {
+      this.refreshBookmarks(state);
+    }
+
+    if (state.loading) {
+      return h('div.spinner-container', h('div.spinner'));
+    }
+
+    return h('ul.menu-bookmarks', [
+      state.content.map(b => h('li', [
```

Comment:
I'm not quite sure what the variable `b` represents here. Usually we want to name variables such that it is obvious on first look what the variable is holding.